src/services/types.ts

```diff
@@ -261,11 +261,13 @@ namespace ts {
         /*
          * LS host can optionally implement these methods to support completions for module specifiers.
          * Without these methods, only completions for ambient modules will be provided.
+         *
+         * `readFile` and `fileExists` are now _required_ to properly acquire and setup source files under module: node12+ modes.

```

Comment:
seems like these should be moved to a new section before or after the remaining two